src/Psalm/Internal/Analyzer/FunctionLike/ReturnTypeAnalyzer.php

```diff
@@ -237,6 +237,7 @@ public static function verifyReturnType(
             foreach ($inferred_return_type_parts as $inferred_return_type_part) {
                 if ($inferred_return_type_part->isVoid()) {
                     $unsafe_return_type = true;
+                    break;
```

Comment:
Loops that only flags a variable should break as soon as possible to avoid useless iterations.